core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2138,7 +2143,8 @@ private[spark] class DAGScheduler(
               // guaranteed to be determinate, so the input data of the reducers will not change
               // even if the map tasks are re-tried.
               if (mapStage.isIndeterminate) {
-                val rollingBackStages = abortIndeterminateStageChildren(mapStage)._2
+                val stagesToRollback = collectSucceedingStages(mapStage)
+                val rollingBackStages = rollBackStages(stagesToRollback)
```

Comment:
I have refactored `abortIndeterminateStageChildren` into two separate methods:
- collectSucceedingStages
- rollBackStages

This way we do not need returning a tuple anymore and we are not calculating the "All jobs with map stage have been aborted" when it is not needed.